packages/core/lib/commands/init/initSource/truffle-config.js

```diff
@@ -16,12 +16,17 @@
  * public/private key pairs. If you're publishing your code to GitHub make sure you load this
  * phrase from a file you've .gitignored so it doesn't accidentally become public.
  *
+ * If you want to use .env for keeping your secret variables, you'll need to install 'dotenv'
+ * (npm install dotenv). To get started with, add .env file in the root directory and declare
+ * your Infura MNEMONIC and PROJECT_ID variables.
+ * .gitignore is already added with this project with .env in it.
  */
 
+// require('dotenv').config();
 // const HDWalletProvider = require('@truffle/hdwallet-provider');
-//
-// const fs = require('fs');
-// const mnemonic = fs.readFileSync(".secret").toString().trim();
+
+// const mnemonic = process.env.MNEMONIC;
+// const projectId = process.env.PROJECT_ID;
```

Comment:
Since the default we give users is reading from `process.env`, that means we want users to use dotenv right? So I would phrase this as:

_It is recommended to store your secret variables in a_ `.env` _file (which you should include in_ `.gitignore`_). In your project root, run_ `npm i dotenv` _, create_ `.env` _and declare your MNEMONIC and infura PROJECT\_ID variables inside._

Reply:
Also destructure maybe? `const { MNEMONIC, PROJECT_ID } = process.env;`

Reply:
Your suggestions make sense to me. I will change the comments. Thanks @cliffoo.